src/Couchbase.Lite.Shared/API/Sync/ReplicatorProgress.cs

```diff
@@ -71,24 +71,39 @@ public struct ReplicatedDocument
         [CanBeNull]
         public CouchbaseException Error { get; internal set; }
 
+        public Collection Collection { get; }
+
         internal bool IsTransient { get; }
 
         internal C4Error NativeError { get; }
 
-        internal ReplicatedDocument([NotNull]string docID, C4RevisionFlags flags, C4Error error,
+        internal ReplicatedDocument([NotNull] string docID, C4RevisionFlags flags, C4Error error,
+            bool isTransient)
+        {
+            Collection = null;
```

Comment:
It would be ideal if this were not allowed, and instead the default collection could be passed in.

Reply:
I think we already discussed this. After implementation started, this constructor will be removed.